### PR TITLE
[Merged by Bors] - chore: use `Submodule.Quotient`-specific induction lemma

### PR DESCRIPTION
**Pull request by Whysoserioushah**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
Can you update the existing uses of `Quotient.inductionOn` specific to `Submodule.Quotient`?

**Comment by Whysoserioushah** [comment]
> Thanks for the PR! Can you update the existing uses of `Quotient.inductionOn` specific to `Submodule.Quotient`?

Done, thanks :)

File: Mathlib/LinearAlgebra/Isomorphisms.lean [hunk 1]
```diff
@@ -158,9 +158,11 @@ def quotientQuotientEquivQuotient : ((M ⧸ S) ⧸ T.map S.mkQ) ≃ₗ[R] M ⧸
   { quotientQuotientEquivQuotientAux S T h with
     toFun := quotientQuotientEquivQuotientAux S T h
     invFun := mapQ _ _ (mkQ S) (le_comap_map _ _)
-    left_inv := fun x => Quotient.inductionOn' x fun x => Quotient.inductionOn' x fun x =>
+    left_inv := fun x => Submodule.Quotient.induction_on _
+     x fun x => Submodule.Quotient.induction_on _ x fun x =>
       by simp [Quotient.mk''_eq_mk]
```
Comment (Ruben-VandeVelde) [inline Mathlib/LinearAlgebra/Isomorphisms.lean:163]: Please check if `Quotient.mk''_eq_mk` is still used, here and below

Comment (Whysoserioushah) [reply Mathlib/LinearAlgebra/Isomorphisms.lean:163]: You're right it shouldn't be used any more, I'll change, thanks for the heads up!

**Review by YaelDillies — APPROVED** [review]
Thanks! 🚀

maintainer merge

**Comment by b-mehta** [comment]
Thanks!

bors merge

**Comment by Whysoserioushah** [comment]
why is it failing I thought I passed all the tests :(

**Comment by YaelDillies** [comment]
The "check the cache" step failed. Looks like a fluke

**Comment by Whysoserioushah** [comment]
> The "check the cache" step failed. Looks like a fluke

Anything I can do about it?

**Pull request closed without merging**